File: Que 3/Python/Polynomial.py
class Pattern:
    def __init__(self):
       
        self.l = list()
        self.calc = 0
        self.lower_bound = int()
        self.upper_bound = int()
        self.incr = int()
        self.m = 55
        self.c = '-'
        self.input()

    def input(self):
        self.n = int(input("  Enter the degree of Polynomial:"))
        print(self.c*self.m)
        for i in range(self.n+1):
            temp = int(input(f"  -> Coeff {i + 1}: "))
            self.l.append(temp)
        
        print(self.c*self.m)

        self.lower_bound = int(input("  Provide lower bound for x: "))
        self.upper_bound = int(input("  Provide upper bound for x: "))
        self.incr = int(input("  Provide number of steps, you wish to increment x: "))
        self.l.reverse()
        
        print(self.c*self.m)
        self.print()
        print(self.c*self.m)

    
    def print(self):
        for i in range(self.lower_bound, self.upper_bound, self.incr):
            calc = 0
            print("| ", end="")
            for index, coeff in enumerate(self.l):
                calc += (i ** index) * coeff
            print("*"*int(calc))

def main():
    m = 55
    c = '-'
    print(c*m)
    print("|" + " "*13 + "POLYNOMIAL FUNCTOIN PROBLEM" + " "*13 + "|")
    print(c*m)
    print("|\t  Provide Polynomial Function Details:\t      |")
    print(c*m)


if __name__ == '__main__':
    main()
    Pattern()

File: Que 3/C/Polynomial.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

#define MAX 3

void polynomial_func(int, int [], int, int , int);
void _print_(char, int);

// This Function prints the characters in the row continously according to the given number.
void _print_(char a, int calc){
    for(int k=0; k<calc; k++){
        printf("%c", a);
    }
    printf("\n");
}

// This Function provide a value from( lower bound to upper bound ) to the polynomial expression that generates a value. 
// that value is passsed to the _print_() function to print the '*' Pattern.
void polynomial_func(int n, int arr[], int lower_bound, int upper_bound, int incr){
    
    int calc;

    for(int i=lower_bound; i<=upper_bound; i = i + incr){

        calc = 0;
        printf("| ");

        for(int j=n; j>=0; j--){
            calc += pow(i, j) * arr[n - j];                 // Calculating the value of the polynomial function acc. to the Inputs given by the user.
        }
        _print_('*', calc);
    }
}


int main(){

    int n, arr[MAX], lower_bound, upper_bound, incr;
    int m = 57;
    
    printf("\n");

    _print_('-', m);
    printf("|\t\tPOLYNOMIAL FUNCTION PROBLEM\t\t|\n");
    _print_('-', m);
    
    printf("|\t    Provide Polynomial Function Details:\t|\n");
    _print_('-', m);

    printf("  Enter degree of Polynomial (MAX=3): ");               // Taking the degree of the polynomial function as input from the user.
    scanf("%d", &n);
    _print_('-', m);

    for(int i=0; i<=n ; i++){
        printf("  ->  coeff %d: ", i + 1);
        scanf("%d", &arr[i]);
    }

    _print_('-', m);
    printf("  Provide lower bound for x: ");
    scanf("%d", &lower_bound);
    
    printf("  Provide upper bound for x: ");
    scanf("%d", &upper_bound);

    printf("  Provide number of steps, you wish to increment x: ");
    scanf("%d", &incr);

    _print_('-', m);
    printf("\n");

    // Calling print_pattern function.
    polynomial_func(n, arr, lower_bound, upper_bound, incr);              
    
    printf("\n");
    _print_('-', m);
    printf("\n");
}